packages/nimble-components/src/table/index.ts

```diff
@@ -259,45 +290,105 @@ export class Table<
     private updateTableOptions(
         updatedOptions: Partial<TanStackTableOptionsResolved<TData>>
     ): void {
-        this.options = { ...this.options, ...updatedOptions };
-        this.update(this.table.initialState);
+        this.options = {
+            ...this.options,
+            ...updatedOptions,
+            state: { ...this.options.state, ...updatedOptions.state }
+        };
+        this.table.setOptions(this.options);
         this.refreshRows();
     }
 
-    private readonly update = (state: TanStackTableState): void => {
-        this.table.setOptions(prev => ({
-            ...prev,
-            ...this.options,
-            state,
-            onStateChange: (updater: unknown) => {
-                const updatedState = typeof updater === 'function'
-                    ? (updater(state) as TanStackTableState)
-                    : (updater as TanStackTableState);
-                this.update(updatedState);
+    private setSortState(): void {
+        const sortedColumns = this.columns
+            .filter(
```

Comment:
nit: you could consider adding a private helper function like `getColumnsParticipatingInSorting()` to reduce the duplication between this code and `validateColumnSortIndices()`